ext/json/json_encoder.c

```diff
@@ -528,6 +528,26 @@ static int php_json_encode_serializable_object(smart_str *buf, zval *val, int op
 }
 /* }}} */
 
+static int php_json_encode_serializable_enum(smart_str *buf, zval *val, int options, php_json_encoder *encoder)
+{
+	zend_class_entry *ce = Z_OBJCE_P(val);
+	if (ce->enum_scalar_type == IS_UNDEF) {
+		encoder->error_code = PHP_JSON_ERROR_NON_SCALAR_ENUM;
+		smart_str_appendc(buf, '0');
+		return FAILURE;
+	}
+
+	zend_string *value_string = zend_string_init("value", strlen("value"), 0);
```

Comment:
nit: Can this and the remaining calls to `zend_string_init("value"...` be replaced with uses of `ZSTR_KNOWN(ZEND_STR_VALUE)` to avoid the extra allocation?

Reply:
Might want to add a zend_enum_ API for fetching the value efficiently.